### ISSUE TITLE
typescript/await-thenable: false positive on await using when file is outside tsconfig include

### ISSUE THREAD

**Issue by art-shen** [oxc-project/tsgolint#830]
I had `typescript/await-thenable` set to `error` and started getting failures on every `await using` in my test files, even though TypeScript compiles them fine. The test files sit outside the tsconfig `include`, and that's the problem trigger.

- Reproducer: https://github.com/art-shen/tsgolint-await-thenable-reproducer
- CI (typecheck passes, lint fails): https://github.com/art-shen/tsgolint-await-thenable-reproducer/actions/runs/23492127237

```bash
bun install
bun run typecheck   # passes
bun run lint        # fails
```

> `tsconfig.json` covers `src/` but not `tests/`. Moving the identical code to `src/` makes the errors disappear. User-defined types that implement `AsyncDisposable` get flagged in the excluded file, even an interface that directly declares `[Symbol.asyncDispose]()`.

```
  x typescript-eslint(await-thenable): Unexpected `await using` of a value that is not async disposable.
    ,-[tests/await-using.ts:14:19]
 13 | export async function withResource() {
 14 |   await using _ = r; // false positive: typescript/await-thenable
    :                   ^
 15 | }
    `----

  x typescript-eslint(await-thenable): Unexpected `await using` of a value that is not async disposable.
    ,-[tests/await-using.ts:23:19]
 22 | export async function withConnection() {
 23 |   await using _ = conn; // false positive: typescript/await-thenable
    :                   ^^^^
 24 | }
    `----

Found 0 warnings and 2 errors.
```

I dug into the source. First suspicion was `bun-types`: `AsyncDisposable` is in `lib.esnext.disposable.d.ts` while `Bun.Subprocess extends AsyncDisposable` is across in `bun.d.ts`, so I wondered if `bun-types` was redeclaring `Symbol.asyncDispose` in a conflicting way. It isn't. Checked `bun-types@1.3.11` directly: no redeclaration, just the `extends`.

Second thought was the TypeScript-Go checker's goroutine pool. It uses four goroutines and assigns symbol IDs incrementally, so I figured execution order might produce mismatched names for the same symbol. Couldn't confirm it, and it didn't explain why `Bun.Subprocess` in the same file produces zero errors anyway. A goroutine race would affect all uses of `AsyncDisposable`, not just user-defined types.

That's what did it. `Bun.Subprocess` is defined in `bun.d.ts`, part of the base program. User-defined interfaces in the linted file are not. **The only thing that matters is whether the type's declaration file ends up as an extra root.**

When a file is outside tsconfig `include`, tsgolint adds it as an extra root to the TypeScript-Go program. That extra-root binding pass is where things go wrong: `getPropertyNameForKnownSymbolName("asyncDispose")` appears to fall back to the no-ID form (`\xFE@asyncDispose`) when the global Symbol constructor isn't yet accessible, so the property gets registered under that name. Later, `GetWellKnownSymbolPropertyOfType` resolves it with an ID suffix (`\xFE@asyncDispose@<n>`) — the exact-name match fails, and the rule concludes the type doesn't implement `AsyncDisposable`.

Versions:
- oxlint `1.56.0`
- oxlint-tsgolint `0.17.3`
- TypeScript `5.7.x`

PR #822 was created to fix that.  Reviewer asked for a standalone reproducer; here it is.

**Comment by camc314** [comment]
This is a configuration issue with your project, not a bug with tsgolint - I will write up a more detailed explanation soon

**Comment by art-shen** [comment]
@camc314, yeah, seems so.

But please consider the final shape of #822 — it can provide a better developer experience anyway.  Because current behavior silently hides the problem's origin.

If the `tsgolint` handles the files (as it does now), I believe it should either handle them well or clearly show an error for files with such a misconfiguration. wdyt?

I've updated the #822 and the PR description. Verified the fix in the reproducer. Please check it out.

**Comment by camc314** [comment]
Hi @art-shen , apologies for the delay.

Since your root `tsconfig` does not include the test files, it means that the `Bun` types are not automatically injected.

I have pushed a commit [here](https://github.com/camc314/tsgolint-await-thenable-reproducer/commit/9d3d3745d13a09429c4887ee5f93217fc40942ba) demonstraighting how to configure TS to get it working as expected.

> If the tsgolint handles the files (as it does now), I believe it should either handle them well or clearly show an error for files with such a misconfiguration. wdyt?

Oxlint is behaving as expected, running with `--type-check` clearly highlights the issue, explaining that there are unresolved imports which can cause cascading errors.

Closing this out as not planned, please reach out if you have any other issues - thanks
